### PR TITLE
Re-order members of RemoteSettingsConfig UDL dict [firefox-android: main] [firefox-ios: main]

### PR DESCRIPTION
**Pull request by brennie**
The order of dictionary members matters to uniffi when they are used to generate arguments for Python functions. In this case, the order was generating a method signature with defaulted kwargs before undefaulted args, which is a syntax error. Re-arranging these members will result in generating a valid Python method signature. The Swift bindings also need updating, because Swift enforces argument order for named arguments, but Kotlin does not.

Fixes bug 1874030

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/remote_settings/src/remote_settings.udl [hunk 1]
```diff
@@ -8,9 +8,9 @@ typedef string RsJsonObject;
 namespace remote_settings {};
 
 dictionary RemoteSettingsConfig {
-    string? server_url = null;
-    string? bucket_name = null;
     string collection_name;
+    string? bucket_name = null;
+    string? server_url = null;
```
Comment (jhugman) [inline components/remote_settings/src/remote_settings.udl:13]: Nit: the URL is more significant than the `bucket_name` or `collection_name`; would you consider having the url before the bucket?

**Review by jhugman — APPROVED** [review]
This requires a Breaking Change in the CHANGELOG.md; there may not be any changes in Firefox for iOS, or Android, but we should still mark it as so.

File: components/remote_settings/src/remote_settings.udl (hunk 1 above)
Comment (brennie) [reply components/remote_settings/src/remote_settings.udl:13]: It is, but its also probably the most likely to be defaulted vs the other two. I think either this order or the previous order with no default value support are the most correct options. I'm going to keep this as is.

**Comment by mhammond** [comment]
IIUC, this is a breaking change for suggest in desktop.

**Pull request merged**